tls/s2n_record_read.c

```diff
@@ -38,11 +38,18 @@ int s2n_sslv2_record_header_parse(
 
     S2N_ERROR_IF(s2n_stuffer_data_available(in) < S2N_TLS_RECORD_HEADER_LENGTH, S2N_ERR_BAD_MESSAGE);
```

Comment:
While you're here... Can we get rid of another `S2N_ERROR_IF` 😄 

```suggestion
    POSIX_ENSURE(s2n_stuffer_data_available(in) >= S2N_TLS_RECORD_HEADER_LENGTH, S2N_ERR_BAD_MESSAGE);
```

Reply:
Also, I'm wondering if we should rename the local `in` variable to `header_in`... I was trying to understand why we are using `in` to process a header until i realized it wasn't `conn->in`, but a local alias of `header_in`.